docs/configuration/http_filters/lua_filter.rst

```diff
@@ -0,0 +1,325 @@
+the next filter in the header chain. For example, they can be modified after an *httpCall()* or
+after a *body()* call returns. The script will fail if the headers are modified in any other
+situation.
+
+Returns a :ref:`header object <config_http_filters_lua_header_wrapper>`.
+
+body()
+^^^^^^
+
+.. code-block:: lua
+
+  body = handle:body()
+
+Returns the stream's body. This call will cause Envoy to yield the script until the entire body
+has been buffered. Note that all buffering must adhere to the flow control policies in place.
+Envoy will not buffer more data than is allowed by the connection manager.
+
+Returns a :ref:`buffer object <config_http_filters_lua_buffer_wrapper>`.
+
+bodyChunks()
+^^^^^^^^^^^^
+
+.. code-block:: lua
+
+  iterator = handle:bodyChunks()
+
+Returns an iterator that can be used to iterate through all received body chunks as they arrive.
+Envoy will yield rthe script in between chunks, but *will not buffer* them. This can be used by
+a script to inspect data as it is streaming by.
+
+.. code-block:: lua
+
+  for chunk in request_handle:bodyChunks() do
+    request_handle:log(0, chunk:byteSize())
+  end
+
+Each chunk the iterator returns is a :ref:`buffer object <config_http_filters_lua_buffer_wrapper>`.
+
+trailers()
+^^^^^^^^^^
+
+.. code-block:: lua
+
+  trailers = handle:trailers()
+
+Returns the stream's trailers. May return nil if there are no trailers. The trailers may be
+modified before they are sent to the next filter.
+
+Returns a :ref:`header object <config_http_filters_lua_header_wrapper>`.
+
+log()
+^^^^^
+
+.. code-block:: lua
+
+  handle:log(level, message)
+
+Logs a message using Envoy's application logging. *level* is an integer in the range 0 - 5 which
+maps to the spdlog levels *trace* - *critical*. *message* is a string to log.
+
+httpCall()
+^^^^^^^^^^
+
+.. code-block:: lua
+
+  headers, body = handle:httpCall(cluster, headers, body, timeout)
+
+Makes an HTTP call to an upstream host. Envoy will yield the script until the call completes or
+has an error. *cluster* is a string which maps to a configured cluster manager cluster. *headers*
+is a table of key/value pairs to send. Note that the *:method*, *:path*, and *:authority* headers
+must be set. *body* is an optional string of body data to send. *timeout* is an integer that
+specifies the call timeout in milliseconds.
+
+Returns *headers* which is a table of response headers. Returns *body* which is the string response
+body. May be nil if there is no body.
+
+respond()
+^^^^^^^^^^
+
+.. code-block:: lua
+
+  handle:respond(headers, body)
+
+Respond immediately and do not continue further filter iteration. This call is *only valid in
+the request flow*. *headers* is a table of key/value pairs to send. Note that the *:status* header
+must be set. *body* is a string and supplies the optional response body. May be nil.
+
+.. _config_http_filters_lua_header_wrapper:
+
+Header object API
```

Comment:
Copy the https://daurnimator.github.io/lua-http/0.2/#http.headers api.

Reply:
Was this considered?

Reply:
I took a look. For now I'm skipping it as am trying to provide a very simple initial offering. We can reconsider in the future depending on demand.

Reply:
It's just as simple. It is just different method names and some slightly different functionality.
This is the sort of class that you'll have huge issues changing in future due to backwards compatibility concerns, so get it right on the first go!